### ISSUE TITLE
Flakey wifi scanning

### ISSUE THREAD

**Issue by IMbackK** [maemo-leste/bugtracker#554]
Right now there seam to be some issues regarding scanning for wifi aps. Scanns are scheduled by icd but the results, which do appear in wpa_supplicant dont always make it to the ui and/or the icd wifi plugin.
Possibly multiple issues with similar symptoms exist

One way to reproduce this:

1. open the connection dialog
2. select am encrypted network that you have never connected to
3. cancel at the psk entry dialg
4. open connection dialog
5. observe that now the connection dialog is borked it will never (or at least not for a long time) show any connections

**Comment by MerlijnWajer** [comment]
I see this on my droid:

```
Jul  3 18:20:18 localhost icd2 0.98[3437]: scan start for type 'WLAN_INFRA'
Jul  3 18:20:18 localhost icd2 0.98[3437]: '[SCAN]' type 'WLAN_INFRA' ICD_STATUS_CHANGED_SIG with status SCAN_START, '' sent to '(null)'
Jul  3 18:20:18 localhost icd2 0.98[3437]: scan start for type 'WLAN_ADHOC'
Jul  3 18:20:18 localhost icd2 0.98[3437]: '[SCAN]' type 'WLAN_ADHOC' ICD_STATUS_CHANGED_SIG with status SCAN_START, '' sent to '(null)'
Jul  3 18:20:18 localhost icd2 0.98[3437]: libicd-network-wpasupplicant: wlan_start_search
Jul  3 18:20:18 localhost icd2 0.98[3437]: libicd-network-wpasupplicant: Starting scan failed.
Jul  3 18:20:18 localhost icd2 0.98[3437]: libicd-network-wpasupplicant: wlan_scan_timeout
Jul  3 18:20:18 localhost icd2 0.98[3437]: module 'libicd_network_wpasupplicant.so' scan completed
Jul  3 18:20:18 localhost icd2 0.98[3437]: scan stop for type 'WLAN_INFRA'
Jul  3 18:20:18 localhost icd2 0.98[3437]: '[SCAN]' type 'WLAN_INFRA' ICD_STATUS_CHANGED_SIG with status SCAN_STOP, '' sent to '(null)'
Jul  3 18:20:18 localhost icd2 0.98[3437]: scan stop for type 'WLAN_ADHOC'
Jul  3 18:20:18 localhost icd2 0.98[3437]: '[SCAN]' type 'WLAN_ADHOC' ICD_STATUS_CHANGED_SIG with status SCAN_STOP, '' sent to '(null)'
Jul  3 18:20:18 localhost icd2 0.98[3437]: libicd-network-wpasupplicant: on_scan_done. params: (true,)
Jul  3 18:20:18 localhost icd2 0.98[3437]: libicd-network-wpasupplicant: SCAN DONE
```

And on `wpa_cli` the scans get started, but the results are not read by the plugin. Seems quite fixable.

This was done using the new connection setup, and hitting Yes for "do you want to scan?" , btw.

**Comment by MerlijnWajer** [comment]
I think this should be fixed in version 0.1.8 in beowulf-devel. Please re-open if it's not fixed.
